pkg/monitoring/source.go

```diff
@@ -24,3 +24,8 @@ type SourceFactory interface {
 	// GetType should return a namespace for all the source instances produced by this factory.
 	GetType() string
 }
```

Comment:
wouldn't it be cleaner to have a separate interface?

otherwise, every existing source would need to implement a new method even if it's not relevant?